R/derive_var_relative_flag.R

```diff
@@ -0,0 +1,179 @@
+#' Flag Observations Before or After a Condition is Fulfilled
+#'
+#' Flag all observations before or after the observation where a specified
+#' condition is fulfilled for each by group. For example, the function could be
+#' called to flag for each subject all observations before the first disease
+#' progression.
+#'
+#' @param dataset Input dataset
+#'
+#'   The variables specified by the `order` and the `by_vars` argument are
+#'   expected.
+#'
+#' @param by_vars Grouping variables
+#'
+#'   *Permitted Values:* list of variables created by `vars()`
+#'
+#' @param order Sort order
+#'
+#'   Within each by group the observations are ordered by the specified order.
+#'
+#'   *Permitted Values:* list of variables or `desc(<variable>)` function calls
+#'   created by `vars()`, e.g., `vars(ADT, desc(AVAL))`
+#'
+#' @param new_var New variable
+#'
+#'   The variable is added to the input dataset and set to `"Y"` for all
+#'   observations before or after the condition is fulfilled. For all other
+#'   observations it is set to `NA`.
+#'
+#' @param condition Condition for Reference Observation
+#'
+#'   The specified condition determines the reference observation. In the output
+#'   dataset all observations before or after (`selection` argument)
+#'   the reference observation are flagged.
+#'
+#' @param mode Selection mode (first or last)
+#'
+#'   If `"first"` is specified, for each by group the observations before or
+#'   after (`selection` argument) the observation where the condition
+#'   (`condition` argument) is fulfilled the *first* time is flagged in the
+#'   output dataset. If `"last"` is specified, for each by group the
+#'   observations before or after (`selection` argument) the observation where
+#'   the condition (`condition` argument) is fulfilled the *last* time is
+#'   flagged in the output dataset.
+#'
+#'   *Permitted Values:* `"first"`, `"last"`
+#'
+#' @param selection Flag observations before or after the reference observation?
+#'
+#'   *Permitted Values:* `"before"`, `"after"`
+#'
+#' @param inclusive Flag the reference observation?
+#'
+#'   *Permitted Values:* `TRUE`, `FALSE`
+#'
+#' @param flag_no_ref_groups Should by groups without reference observation be flagged?
+#'
+#'   *Permitted Values:* `TRUE`, `FALSE`
+#'
+#' @param check_type Check uniqueness?
```

Comment:
i was comparing the arguments against `filter_relative.R` and spotted there that we have "*Default:* `"none"`" whereas the actual default is "warning". if you were to remove all the default lines in the argument documentation for `filter_relative.R` then it will make these functions more consistent and accurate, plus follows similar thinking to what we were implementing recently for admiralonco

Reply:
Good catch, I have removed the default from the documentation.